crates/rethnet/src/lib.rs

```diff
@@ -1,18 +1,68 @@
 use std::ffi::OsString;
+use std::net::{IpAddr, Ipv4Addr, SocketAddr};
 
-use clap::{Parser, Subcommand};
+use anyhow::anyhow;
+use clap::{Args, Parser, Subcommand};
+use hashbrown::HashMap;
+use tracing::{event, Level};
+
+use rethnet_eth::{Address, U256};
+use rethnet_evm::{AccountInfo, KECCAK_EMPTY};
+use rethnet_rpc_server::{Config, RpcForkConfig, RpcHardhatNetworkConfig};
 
 #[derive(Parser)]
 #[clap(author, version, about, long_about = None)]
-struct Args {
+struct Cli {
     #[clap(subcommand)]
     command: Command,
 }
 
 #[derive(Subcommand)]
 #[allow(clippy::large_enum_variant)]
 enum Command {
-    Start,
+    Node(NodeArgs),
+}
+
+#[derive(Args)]
+struct NodeArgs {
+    #[clap(long, default_value = "127.0.0.1")]
+    host: String,
+    #[clap(long, default_value = "8545")]
+    port: u16,
+    #[clap(long)]
+    fork_url: Option<String>,
+    #[clap(long)]
+    fork_block_number: Option<usize>,
+}
+
+impl TryFrom<NodeArgs> for Config {
+    type Error = anyhow::Error;
+
+    fn try_from(node_args: NodeArgs) -> Result<Config, Self::Error> {
+        Ok(Config {
+            address: match node_args.host.as_str() {
+                "127.0.0.1" => {
+                    SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), node_args.port)
+                }
+                _ => format!("{}:{}", node_args.host, node_args.port).parse()?,
+            },
```

Comment:
```suggestion
        let ip = node_args.host.map_or_else(
            || Ok(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)))
            |host| IpAddr::from_str(&host)
        )?;

        Ok(Config {
            address: SocketAddr::new(ip, node_args.port),
```

Reply:
This is what I had in mind. We're no longer constructing a temporary string using `format!`, which is expensive, nor parsing the IP address if it wasn't specified

Reply:
Thanks for pushing this further. Re-addressed in https://github.com/NomicFoundation/hardhat/pull/4057/commits/8df18fc042d806758d2d12f46b7240dbf22bbc66

I did it a little different than your suggestion, because `node_args.host` actually isn't an option, since if it was left unspecified on the command line, at this point it has been resolved to it's default value. But my latest revision eliminates the `format!()` as well as the parsing of the port.

Reply:
Maybe this implicit detail was not clear based on my suggested code, but my point was that, from the user perspective, there is no difference between:

1. a manual mention of [default: 127.0.0.1]
```
host: Option<String>
```

and

2.
```
#[clap(..., default = "127.0.0.1")]
host: String,
```

Internally, `Clap` with just do the `unwrap_or_default` for you. Converting to a string and then having to do a `match` to detect a string equality is just a waste when we know that the default value when a `None` occurs should be a `const` constructed `IpV4Address`

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/4057/commits/8f61a479918d8edb63aacb5064ca161fcb0bc9a6